templates/challenge.php

```diff
@@ -13,5 +13,11 @@
 </form>
 
 <img src="<?php print_unescaped(image_path('twofactor_u2f', 'app.svg')); ?>">
-<p><?php p($l->t('Please plug in your U2F device and press the device button to authorize.')) ?></p>
+<p id="u2f-info">
+	<?php p($l->t('Please plug in your U2F device and press the device button to authorize.')) ?>
+</p>
+<p id="u2f-error"
+   style="display: none">
+	<strong><?php p($l->t('An error occurred. Please try again.')) ?></strong>
+</p>
 <p><em><?php p($l->t('Chrome is the only browser that supports U2F devices. You need to install the "U2F Support Add-on" on Firefox to use U2F.')) ?></em></p>
```

Comment:
Right, good point. However, I'm not a big fan of opera and apparently only about one percent of internet users is using opera. Not sure if it's worth to mention here as it has also not been tested on a opera browser yet.